app/models/topic.rb

```diff
@@ -855,22 +855,31 @@ def self.reset_all_highest!
         FROM posts
         WHERE deleted_at IS NULL AND post_type <> 4
         GROUP BY topic_id
-      )
+      ),
+      Z as (
+        SELECT topic_id,
+               SUM(COALESCE(posts.word_count, 0)) word_count
+        FROM posts
+        WHERE deleted_at IS NULL AND post_type <> 4
+        GROUP BY topic_id
+      )       
       UPDATE topics
       SET
         highest_staff_post_number = X.highest_post_number,
         highest_post_number = Y.highest_post_number,
         last_posted_at = Y.last_posted_at,
-        posts_count = Y.posts_count
-      FROM X, Y
+        posts_count = Y.posts_count,
+        word_count = Z.word_count
+      FROM X, Y, Z
       WHERE
         topics.archetype <> 'private_message' AND
         X.topic_id = topics.id AND
         Y.topic_id = topics.id AND (
           topics.highest_staff_post_number <> X.highest_post_number OR
           topics.highest_post_number <> Y.highest_post_number OR
           topics.last_posted_at <> Y.last_posted_at OR
-          topics.posts_count <> Y.posts_count
+          topics.posts_count <> Y.posts_count OR
+          topics.word_count <> Z.word_count
```

Comment:
Probably worth adding a line for `word_count` to the tests that cover `Topic.reset_all_highest!`.

Reply:
I did in https://github.com/discourse/discourse/pull/27065/files#diff-8e5efa4a49d72491e785af544164376095eebc1341924e2445dc32c3ff29d788R1192-R1217, didn't I? 🤔

Reply:
Yes, sorry! My mistake. 

You did in https://github.com/discourse/discourse/blob/c8601a5784cc0866060a960275d49562cd5bcd90/spec/models/topic_spec.rb#L2877